### PR TITLE
 Implement Nameable, Decorator, and Decorator Classes

### PR DESCRIPTION
**Pull request by DevaWinner**
In this pull request, I introduce new classes and decorators to enhance the functionality of the classes. The decorators allow for flexible modification of the correct_name method's behavior without modifying the original Person class.

**Changes Made:**

- Created a `Nameable` class with the `correct_name` method to raise `NotImplementedError`.
- Modified the `Person` class to inherit from `Nameable` and implemented the `correct_name` method to return the person's name.
- Implemented the base `decorator` class, which inherits from `Nameable` and wraps a `nameable` object to delegate the `correct_name` method.
- Created `capitalize-decorator`, a subclass of `decorator`, which capitalizes the output of `@nameable.correct_name`.
- Created `decorator--trimmer`, another subclass of `decorator`, which trims the output of `@nameable.correct_name` to a maximum of 10 characters.

The code has been organized into separate files for each class to maintain code modularity and readability.

Please review and approve this pull request, and let me know if any correction is needed.

**Usage:**

```ruby
person = Person.new(22, 'maximilianus')
puts "Original name: #{person.correct_name}"

capitalized_person = CapitalizeDecorator.new(person)
puts "Capitalized name: #{capitalized_person.correct_name}"

capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
puts "Capitalized and trimmed name: #{capitalized_trimmed_person.correct_name}"

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
```
Comment (Owusu-Desmond) [inline decorator.rb]: - [ ] From the project requirement, this `correct_name` should capitalize the first letter of the given name so I think you should method return `@nameable.correct_name.capitalize` to get this functionally done. :blush:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @DevaWinner , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (DevaWinner) [reply decorator.rb]: Thank you. I have corrected that.

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (OyePriscilla) [inline decorator.rb:12]: - [opional] Please kindly take note that you're not suppose to capitalize the base decorator, the previous reviewer made  a mistake I think. Please ensure that you correct this for your project to fulfil the aim of this milestone. 🙏 

Code review

**Review by OyePriscilla — APPROVED** [review]
Hi @DevaWinner 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
Implemented composition (as an example of the design pattern). ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**